dcs_deploy.py

```diff
@@ -161,14 +186,64 @@ def check_status(self, group = None):
                 break
     
     def get_status(self, group = None):
-        #check if configuration was deleled. If yes, reload default configuration
+        # Check if configuration was deleled. If yes, reload default configuration
         if not os.path.isfile(self.status_file_name):
             self.load()
 
         if group == None:
             group = self.group
         return self.status[group]["status"]
-   
+
+class BatchCounter:
+    def __init__(self, counter_file_path:str, counter_value:int = 0):
+        try:
+            counter_value = int(counter_value)
+        except ValueError:
+                raise ValueError(f"Invalid counter_value: {counter_value}. Must be an integer.")
+        
+        if (counter_value < 1 or counter_value > 254) and (counter_value != 0):
+            print('Invalid counter value! Counter value must be between 1 and 254.')
+            exit()
+        self.counter_file_path = counter_file_path
+
+        # Check if the file exists
+        if not os.path.isfile(self.counter_file_path):
+            # If counter_value is within the desired range, use it; otherwise, default to 1
+            self.counter = counter_value if 1 <= counter_value <= 254 else 1
+            with open(self.counter_file_path, 'w') as file:
+                # Write the initial counter value to the file
+                file.write(str(self.counter))
+                print("Batch counter file created.")
+        else:
+            with open(self.counter_file_path, 'r') as file:
+                stored_value = file.read()
+                try:
+                    stored_value = int(stored_value)
+                    # If counter_value is within the desired range, use it;
+                    # otherwise, use the value read from the file
+                    self.counter = counter_value if 1 <= counter_value <= 254 else stored_value
+                except ValueError:
+                    self.counter = 1
+                    print("Warning: Batch counter file was corrupt or empty, resetting to 1.")
+
+        # This ensures that even if the file exists and counter_value is within the desired range,
+        # the counter is set accordingly and saved to the file.
+        if 1 <= counter_value <= 254:
+            self.save()
+        print(f"Jetson is going to be flashed with ID: {self.counter}.")
+        
+    def increment(self):
+        if self.counter < 254:
+            self.counter += 1
+        else:
+            print("Batch counter has reached its maximum value. Resetting to 1.")
+            self.counter = 1
+        self.save()
+
+    def save(self):
+        with open(self.counter_file_path, 'w') as file:
+            file.write(str(self.counter))
+
```

Comment:
I feel like this could be moved to a separate unit (file) to maintain readability. But it is just a note, not a requirement.

Reply:
I thought of this already. There should be `util` modul with all the commands (https://github.com/airvolute/dcs-deploy/blob/custom_rootfs_and_batch/dcs_deploy.py#L16-L69) and bot helper classes from dcs_deploy.py.

Let's mark it for 2.0 release target - #20

Is it okay?

Reply:
Yes.